init_project.sh

```diff
@@ -39,25 +39,94 @@ function checkoutSourceCodeFromGit()
             git config --global core.eol LF
             git config --global diff.renamelimit 5000
         fi
-        status "Checking out CE repository"
-        git clone ${repository_url_ce} "${magento_ce_dir}" 2> >(logError) > >(log)
-        status "Checking out CE sample data repository"
-        repository_url_ce_sample_data="$(bash "${vagrant_dir}/scripts/get_config_value.sh" "repository_url_ce_sample_data")"
-        git clone ${repository_url_ce_sample_data} "${magento_ce_sample_data_dir}" 2> >(logError) > >(log)
+
+        initMagentoCeGit
+        initMagentoCeSampleGit
+
         # By default EE repository is not specified and EE project is not checked out
         if [[ -n "${repository_url_ee}" ]]; then
-            status "Checking out EE repository"
-            git clone ${repository_url_ee} "${magento_ee_dir}" 2> >(logError) > >(log)
+            initMagentoEeGit
         fi
         # By default EE sample data repository is not specified and EE project is not checked out
         repository_url_ee_sample_data="$(bash "${vagrant_dir}/scripts/get_config_value.sh" "repository_url_ee_sample_data")"
         if [ -n "${repository_url_ee_sample_data}" ]; then
-            status "Checking out EE sample data repository"
-            git clone ${repository_url_ee_sample_data} "${magento_ee_sample_data_dir}" 2> >(logError) > >(log)
+            initMagentoEeSampleGit
         fi
     fi
 }
 
+function initMagentoCeGit()
+{
+    initGitRepository ${repository_url_ce} "CE" ${magento_ce_dir}
```

Comment:
All directories should be wrapped in double quotes, otherwise it is impossible to deploy this vagrant project to a directory which contains spaces in its full path.